File: studio/AuraStudio/Tests/AuraStudioTests/MusicBrainzClientTests.swift
import XCTest
@testable import AuraStudio

/// D-203: `buildQuery` es la pieza que causaba busquedas rotas en
/// silencio con titulos reales (comillas, barras invertidas) -- ver
/// DECISIONS.md. Sin red, verifica solo el armado de la query Lucene.
final class MusicBrainzClientTests: XCTestCase {
    func testPlainTitleAndArtistAreQuoted() {
        let query = MusicBrainzClient.buildQuery(title: "Bohemian Rhapsody", artist: "Queen")
        XCTAssertEqual(query, "recording:\"Bohemian Rhapsody\" AND artist:\"Queen\"")
    }

    func testDoubleQuoteInTitleIsEscapedNotLeftBroken() {
        let query = MusicBrainzClient.buildQuery(title: "Rock \"N\" Roll", artist: nil)
        XCTAssertEqual(query, "recording:\"Rock \\\"N\\\" Roll\"")
    }

    func testBackslashInArtistIsEscaped() {
        let query = MusicBrainzClient.buildQuery(title: nil, artist: "Y\\N")
        XCTAssertEqual(query, "artist:\"Y\\\\N\"")
    }

    func testOnlyTitleOmitsArtistClause() {
        let query = MusicBrainzClient.buildQuery(title: "Yesterday", artist: nil)
        XCTAssertEqual(query, "recording:\"Yesterday\"")
    }
}

File: studio/windows/tests/AuraStudio.Core.Tests/Networking/MusicBrainzClientTests.cs
using AuraStudio.Core.Networking;
using Xunit;

namespace AuraStudio.Core.Tests.Networking;

/// <summary>
/// D-203: <c>buildQuery</c> es la pieza que causaba búsquedas rotas en
/// silencio con títulos reales (comillas, barras invertidas) — ver
/// DECISIONS.md. Sin red, verifica solo el armado de la query Lucene
/// (equivalente de MusicBrainzClientTests.swift).
/// </summary>
public class MusicBrainzClientTests
{
    [Fact]
    public void PlainTitleAndArtistAreQuoted()
    {
        var query = MusicBrainzClient.BuildQuery("Bohemian Rhapsody", "Queen");
        Assert.Equal("recording:\"Bohemian Rhapsody\" AND artist:\"Queen\"", query);
    }

    [Fact]
    public void DoubleQuoteInTitleIsEscapedNotLeftBroken()
    {
        var query = MusicBrainzClient.BuildQuery("Rock \"N\" Roll", null);
        Assert.Equal("recording:\"Rock \\\"N\\\" Roll\"", query);
    }

    [Fact]
    public void BackslashInArtistIsEscaped()
    {
        var query = MusicBrainzClient.BuildQuery(null, "Y\\N");
        Assert.Equal("artist:\"Y\\\\N\"", query);
    }

    [Fact]
    public void OnlyTitleOmitsArtistClause()
    {
        var query = MusicBrainzClient.BuildQuery("Yesterday", null);
        Assert.Equal("recording:\"Yesterday\"", query);
    }
}
